### ISSUE TITLE
[a11y] Consider using focus-visible

### ISSUE THREAD

**Issue by ry5n** [Shopify/polaris#818]
**Scott Vinkle** [on Slack](https://shopify.slack.com/archives/C03T74ZTD/p1521213943000061)
Consider looking at things like `focus-visible` (https://github.com/WICG/focus-visible) or `what-input` (https://github.com/ten1seven/what-input) for keyboard-only focus styles.

This would allow us to not show focus rings for mouse interactions but show them when using the keyboard.

> Especially the `focus-visible` polyfill, then we can remove that when it gets into browsers natively.

**Comment by ry5n** [comment]
We should implement this in order to fix https://github.com/Shopify/polaris-ux/issues/237
